packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -324,8 +486,32 @@ class _CheckboxPainter extends ToggleablePainter {
       ..strokeCap = StrokeCap.round;
   }
 
-  void _drawBox(Canvas canvas, Rect outer, Paint paint, BorderSide? side, bool fill) {
-    if (fill) {
+  void _drawBox(Canvas canvas, Rect outer, Paint paint, BorderSide? side) {
+    // Add dropshadow effect when unselected.
+    // The gradient colors were eyeballed from examples in the HIG docs.
+    //
+    // Since the focus outline is drawn around the outer border,
+    // only clip to the outer border when not in focus.
+    if (value == false && reaction.isDismissed && !isFocused){
+      canvas.clipPath(shape.getOuterPath(outer));
+      canvas.drawPath(shape.getOuterPath(outer), paint);
+
+      // The drop shadow has three layers.
+      final Rect shadowRect = Rect.fromLTRB(
+        outer.left, outer.top, outer.right, outer.top + 3);
+      final LinearGradient topEdgeGradient = LinearGradient(
+        begin: Alignment.topCenter,
+        end: Alignment.bottomCenter,
+        colors: <Color>[
+          CupertinoColors.black.withOpacity(0.1),
+          CupertinoColors.black.withOpacity(0.05),
+        ],
+      );
+      final Paint gradientPaint = Paint()
+        ..shader = topEdgeGradient.createShader(shadowRect);
```

Comment:
I am exploring passing a `LinearGradient` as a parameter to `CupertinoCheckbox` to be used in place of `topEdgeGradient`. Is that a good idea, considering `shadowRect` is independent of `topEdgeGradient` (and used alongside it in drawing the shadow)? Is there a better way to do this (callbacks, etc) or should I just keep it simple and use a `bool`?

Reply:
I've been thinking about this more. I think we don't want to expose too complicated of a parameter, because it gets messy when if we think about the Material check box. It raises the question of whether we should add the equivalent to Material. But we don't provide shadow properties a lot because we generally encourage designing through composability. So we give them an easy way to remove the shadow, and if they want a different one, they wrap the widget with their own shadow. A `bool` could work, but maybe it would be more elegant to provide a `shadowColor`? And they could hide the shadow by making it transparent?

Reply:
Yeah, I have seen shadowColor in the material library. I think it is a pretty common pattern there 👍

Reply:
On further inspection I've found that the shadow in native macOS checkboxes are even smaller and less-layered than previously thought. Considering its relatively minimal visual (and functional) effect, and the complexity of an additional parameter (without a Material equivalent, to boot), it might be worth it to consider sacrificing native fidelity for an easier to use, Material-compatible API in this instance. @MitchellGoodwin @Piinks

Reply:
I'm fine without a shadow. They can wrap them with a shadow if they want one.